src/node_modules/app/components/elementary/ElasticSearchField.svelte

```diff
@@ -176,15 +181,23 @@
 			};
 			return unionEditor;
 		} else if (targetDataType.endsWith('[]')) {
+			let subType;
+			if (typeof simplfiedTypeObject === 'string' && simplfiedTypeObject.endsWith('[]')) {
```

Comment:
(*) I'd define funtions like `const isArrayType = makeEndsWith('[]')` and `isStringType` etc to make thir a bit more readable (should read more like English)